library/ecp.c

```diff
@@ -2287,12 +2287,14 @@ static int ecp_mul_comb( mbedtls_ecp_group *grp, mbedtls_ecp_point *R,
         mbedtls_free( T );
     }
 
-    /* don't free R while in progress in case R == P */
+    int should_free_R = 0;
+    /* prevent caller from using invalid value */
+    should_free_R = ( ret != 0 );
 #if defined(MBEDTLS_ECP_RESTARTABLE)
-    if( ret != MBEDTLS_ERR_ECP_IN_PROGRESS )
+    /* don't free R while in progress in case R == P */
+    should_free_R = should_free_R && ( ret != MBEDTLS_ERR_ECP_IN_PROGRESS );
```

Comment:
In this instance I think
```
if( ret != MBEDTLS_ERR_ECP_IN_PROGRESS )
     should_free_R = 0;
```
would really be more readable. (Gilles made that point more generally, and for most instances both ways look OK enough to me, but in this instance I really prefer using an `if`.)

Reply:
Sounds good to me, but I believe the condition should be inverted like so:
```c
if( ret == MBEDTLS_ERR_ECP_IN_PROGRESS )
    should_free_R = 0;
```

Reply:
Absolutely, my mistake, thanks for catching it.